tests/epyccel/test_arrays.py

```diff
@@ -4049,6 +4049,7 @@ def test_array_float_nested_C_array_initialization_3(language):
         pytest.param("python", marks = pytest.mark.python)
     )
 )
+@pytest.mark.skipif(lambda lang: lang == 'fortran', reason="List comprehension is not supported in Fortran")
```

Comment:
The missing C support seemed very minimal. Is there any reason not to make the same minimal changes to fix Fortran?